gradle-plugin/openshift/src/test/java/org/eclipse/jkube/gradle/plugin/task/OpenShiftUndeployTaskTest.java

```diff
@@ -0,0 +1,129 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockConstruction;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class OpenShiftUndeployTaskTest {
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  private MockedConstruction<DefaultTask> defaultTaskMockedConstruction;
+  private MockedConstruction<OpenshiftUndeployService> openshiftUndeployServiceMockedConstruction;
+  private MockedConstruction<ClusterAccess> clusterAccessMockedConstruction;
+  private Project project;
+  private boolean isOffline;
+
+  @Before
+  public void setUp() throws IOException {
+    project = mock(Project.class, RETURNS_DEEP_STUBS);
+    final OpenShiftClient openShiftClient = mock(OpenShiftClient.class);
+    when(openShiftClient.getMasterUrl()).thenReturn(new URL("http://api.openshiftapps.com:6443"));
+    when(openShiftClient.isAdaptable(OpenShiftClient.class)).thenReturn(true);
+    defaultTaskMockedConstruction = mockConstruction(DefaultTask.class, (mock, ctx) -> {
+      when(mock.getProject()).thenReturn(project);
+      when(mock.getLogger()).thenReturn(mock(Logger.class));
+    });
+    clusterAccessMockedConstruction = mockConstruction(ClusterAccess.class, (mock, ctx) -> when(mock.createDefaultClient()).thenReturn(openShiftClient));
+    openshiftUndeployServiceMockedConstruction = mockConstruction(OpenshiftUndeployService.class);
+    isOffline = false;
+    final OpenShiftExtension extension = new TestOpenShiftExtension() {
+      @Override
+      public Property<Boolean> getOffline() {
+        return new DefaultProperty<>(Boolean.class).value(isOffline);
+      }
+    };
+    when(project.getName()).thenReturn("test-project");
+    when(project.getProjectDir()).thenReturn(temporaryFolder.getRoot());
+    when(project.getBuildDir()).thenReturn(temporaryFolder.newFolder("build"));
+    when(project.getConfigurations().stream()).thenAnswer(i -> Stream.empty());
+    when(project.getBuildscript().getConfigurations().stream()).thenAnswer(i -> Stream.empty());
+    when(project.getExtensions().getByType(OpenShiftExtension.class)).thenReturn(extension);
+    when(project.getConvention().getPlugin(JavaPluginConvention.class)).thenReturn(mock(JavaPluginConvention.class));
+  }
+
+  @After
+  public void tearDown() {
+    defaultTaskMockedConstruction.close();
+    openshiftUndeployServiceMockedConstruction.close();
+    clusterAccessMockedConstruction.close();
+  }
+
+  @Test
+  public void runTask_withOffline_shouldThrowException() {
+    // Given
+    isOffline = true;
+    final OpenShiftUndeployTask undeployTask = new OpenShiftUndeployTask(OpenShiftExtension.class);
+
+    // When
+    IllegalArgumentException illegalStateException = assertThrows(IllegalArgumentException.class, undeployTask::runTask);
+
+    // Then
+    assertThat(illegalStateException.getMessage())
```

Comment:
In most of your PRs you're doing this. Remember AssertJ offers specific assertions for exceptions.

`assertThat(illegalStateException).hasMessage("` 

should be preferred.